Document the new `CUDAQ_MGPU_FUSE` environment variable (#1576)
NVIDIA/cuda-quantum: docs/sphinx/using/backends/simulators.rst

The :code:`nvidia-mgpu` target provides a state vector simulator accelerated with 
the :code:`cuStateVec` library but with support for Multi-Node, Multi-GPU distribution of the 

> state vector. (It can also run on a single GPU.)

Reviewer: ```suggestion
state vector, in addition to a single GPU.
```

  This backend requires an NVIDIA GPU, CUDA runtime libraries, as well as an MPI installation. If you do not have these dependencies installed, you may encounter either an error stating `invalid simulator requested` (missing CUDA libraries), or an error along the lines of `failed to launch kernel` (missing MPI installation). See the section :ref:`dependencies-and-compatibility` for more information about how to install dependencies.

The :code:`nvidia-mgpu` backend has additional performance improvements to
help reduce your simulation runtimes, even on a single GPU. One of the

> performance improvements is to fuse multiple gates together during runtime. By

Reviewer: Maybe explain in a few words what does "fuse multiple gates together during runtime" mean

Author: Updated in 88f2272a0
